engine/src/flutter/shell/platform/darwin/ios/platform_view_ios.mm

```diff
@@ -159,6 +159,19 @@ new PlatformMessageHandlerIos(task_runners.GetPlatformTaskRunner())) {}
   }
 }
 
+// |PlatformView|
+void PlatformViewIOS::SetApplicationLocale(const std::string& locale) {
+  FML_DCHECK(owner_controller_);
+  FML_DCHECK(accessibility_bridge_);
+  if (owner_controller_) {
```

Comment:
Nit: Since the owner_controller_ is an Objective-C object, you don't need the explicit null check. Something like this would be equivalent and more terse.

```c++
owner_controller_.applicationLocale = locale.empty() ? nil : @(locale.data());
```